spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala

```diff
@@ -48,58 +49,38 @@ object DNSSuspiciousConnectsAnalysis {
     * @param logger
     */
   def run(config: SuspiciousConnectsConfig, sparkContext: SparkContext, sqlContext: SQLContext, logger: Logger,
-          inputDNSRecords: DataFrame) = {
+          inputDNSRecords: DataFrame): SuspiciousConnectsAnalysisResults = {
 
 
     logger.info("Starting DNS suspicious connects analysis.")
 
-    val cleanDNSRecords = filterAndSelectCleanDNSRecords(inputDNSRecords)
+    val dnsRecords = filterRecords(inputDNSRecords)
+      .select(InSchema: _*)
+      .na.fill(DefaultQueryClass, Seq(QueryClass))
+      .na.fill(DefaultQueryType, Seq(QueryType))
+      .na.fill(DefaultQueryResponseCode, Seq(QueryResponseCode))
 
-    val scoredDNSRecords = scoreDNSRecords(cleanDNSRecords, config, sparkContext, sqlContext, logger)
+    logger.info("Fitting probabilistic model to data")
+    val model =
+      DNSSuspiciousConnectsModel.trainModel(sparkContext, sqlContext, logger, config, dnsRecords)
 
-    val filteredDNSRecords = filterScoredDNSRecords(scoredDNSRecords, config.threshold)
+    logger.info("Identifying outliers")
+    val scoredDNSRecords = model.score(sparkContext, sqlContext, dnsRecords, config.userDomain)
 
-    val orderedDNSRecords = filteredDNSRecords.orderBy(Score)
+    val filteredScored = filterScoredRecords(scoredDNSRecords, config.threshold).orderBy(Score)
```

Comment:
I removed that but then it got reverted when I did rebase with master. Will change again, thanks.